packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -278,8 +278,12 @@ function warnOnHookMismatchInDev(currentHookName: HookType) {
 function throwInvalidHookError() {
   invariant(
     false,
-    'Hooks can only be called inside the body of a function component. ' +
-      '(https://fb.me/react-invalid-hook-call)',
+    'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for' +
+      ' one of the following reasons:\n' +
+      '1. You might have mismatching versions of React and React DOM\n' +
```

Comment:
Technically this can fire in RN too, so the message should probably say "of React and the renderer (such as React DOM)"

Reply:
I copied from the React docs. Should we also maybe update those as well?

Reply:
Will open a different PR there.

Reply:
In the React docs it's more acceptable because React docs in generally use DOM for examples. But seeing "React DOM" in your native console is weird.